### ISSUE TITLE
Stop training via test set error

### ISSUE THREAD

**Issue by bcipolli** [guruucsd/DifferentialEncoding#67]
Currently, the code stops training based on training set error. Most of the time this doesn't matter--our train and test sets are the same, or we use heavy regularization.

But, good to tweak things to do it "right", and avoid overtraining natural images, or cases (like the Kitterle experiment) that have a test set.
